### PR TITLE
Provider a user option to enable/disable the 'Use auto property' analyzer

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://developercommunity.visualstudio.com/content/problem/23029/no-option-to-disable-ide0032-code-style-warning.html

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by heejaechang** [comment]
sorry, forgot about this bug. let me work on this after my IOperation thing I am doing for thursday.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest please

**Comment by CyrusNajmabadi** [comment]
@sharwell @Pilchie @dotnet/roslyn-ide if you take https://github.com/dotnet/roslyn/pull/23237 then you probably want to take this as well.

https://github.com/dotnet/roslyn/pull/23237 Will make "use auto property" show up a lot more often (since it's not gated on FSA).  This means there will likely be people who want to disable the feature.

File: src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs [hunk 1]
```diff
@@ -597,6 +620,7 @@ public Customer(string value)
 ";
 
         private const string s_preferExpressionBodyForMethods = @"
+>>>>>>> master
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs]: yikes.

**Comment by CyrusNajmabadi** [comment]
This is now a followup PR to: https://github.com/dotnet/roslyn/pull/23237

File: src/VisualStudio/Core/Def/ServicesVSResources.resx [hunk 2]
```diff
@@ -900,7 +903,7 @@ Additional information: {1}</value>
   <data name="VisualStudioWorkspace_TryApplyChanges_cannot_be_called_from_a_background_thread" xml:space="preserve">
     <value>VisualStudioWorkspace.TryApplyChanges cannot be called from a background thread.</value>
   </data>
-  <data name="prefer_auto_properties" xml:space="preserve">
+  <data name="prefer_auto_properties_2" xml:space="preserve">
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/ServicesVSResources.resx]: :bulb: I found the names here confusing. Consider renaming the original `prefer_auto_properties` to `codegen_prefer_auto_properties`, and adding the `codegen_` prefix to `prefer_throwing_properties`. Then make `Prefer_auto_properties` simply refer to the new string.

File: src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs [hunk 3]
```diff
@@ -46,10 +46,15 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             foreach (var diagnostic in context.Diagnostics)
             {
+                var priority = diagnostic.Properties.ContainsKey("LowPriority")
```
Comment (sharwell) [inline src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs]: ❓ Is there a reason this doesn't use the approach of `UseExpressionBodyCodeFixProvider` (i.e. setting to low priority when the diagnostic severity is hidden)?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs]: nope.  that seems reasonable :)

File: src/VisualStudio/Core/Def/ServicesVSResources.resx (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/ServicesVSResources.resx]: sounds good.

File: src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/UseAutoProperty/AbstractUseAutoPropertyCodeFixProvider.cs]: done.

File: src/VisualStudio/Core/Def/ServicesVSResources.resx (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/ServicesVSResources.resx]: done.

**Comment by CyrusNajmabadi** [comment]
@sharwell Ready for you.

**Comment by sharwell** [comment]
@Pilchie for ask mode

**Comment by madskristensen** [comment]
@kuhlenh opportunity for PR to https://github.com/madskristensen/EditorConfigLanguage/blob/master/src/Schema/EditorConfig.json

**Comment by logiclrd** [comment]
So if I'm running Visual Studio 2017 Version 15.6.0 Preview 5.0, I should now have this feature? How exactly do I make use of it? Did an option get added to the settings dialog, or is it accessible only via `.editorconfig`? How exactly is this option enabled in `.editorconfig`? Thanks!

**Comment by logiclrd** [comment]
Never mind me, I just found it in the settings dialog simply by searching for "Prefer". However, I'm still not sure how one activates the function via `.editorconfig`. Thanks again :-)

**Comment by sharwell** [comment]
@logiclrd The .editorconfig option is `dotnet_style_prefer_auto_properties` 😄

**Pull request merged**